cms/djangoapps/api/v1/serializers/course_runs.py

```diff
@@ -168,29 +168,33 @@ def create(self, validated_data):
 class CourseRunRerunSerializer(CourseRunSerializerCommonFieldsMixin, CourseRunTeamSerializerMixin,
                                serializers.Serializer):
     title = serializers.CharField(source='display_name', required=False)
+    number = serializers.CharField(source='id.course', required=False)
     run = serializers.CharField(source='id.run')
 
-    def validate_run(self, value):
+    def validate(self, attrs):
         course_run_key = self.instance.id
+        _id = attrs.get('id')
+        number = _id.get('course') or course_run_key.course
+        run = _id['run']
         store = modulestore()
         with store.default_store('split'):
-            new_course_run_key = store.make_course_key(course_run_key.org, course_run_key.course, value)
+            new_course_run_key = store.make_course_key(course_run_key.org, number, run)
```

Comment:
Do we need to catch an `InvalidKeyError` to check the `number` is well formatted?

Reply:
I suppose we should. Weird that we didn't have that check previously, but I guess we may have had the assumption that all the values would be safe. Will add